engine/src/main/java/io/camunda/zeebe/engine/processing/common/ExpressionProcessor.java

```diff
@@ -224,6 +226,28 @@ public ExpressionProcessor(
         .map(EvaluationResult::getList);
   }
 
+  /**
+   * Evaluates the given expression and returns the result as a list of strings.
+   *
+   * @param expression the expression to evaluate
+   * @param scopeKey the scope to load the variables from (a negative key is intended to imply an
+   *     empty variable context)
+   * @return either the evaluation result as a list of regular strings, or a failure if the
+   *     evaluation fails
+   */
+  public Either<Failure, List<String>> evaluateArrayOfStringsExpression(
```

Comment:
:wrench: I would like to see a Unit test or a couple of unit tests just for this method. Could also be a good place to test edge cases like empty lists, null elements in the list (which I think are allowed in Feel), string elements with just a space character, elements in the list which are numbers, scopes ... stuff like that

Reply:
I totally agree. I've added a test below for those few cases we can test with the current implementation. It highlights the problems with the `convertToJson` implementation. I'll discuss it in the other discussion point below.